File: android/src/main/java/org/axen/flutter/glide/common/GlideImagePool.java
package org.axen.flutter.glide.common;

import org.axen.flutter.glide.constant.BoxFit;
import org.axen.flutter.glide.constant.Resource;

import java.util.concurrent.ConcurrentLinkedQueue;

public class GlideImagePool {
    private static final ConcurrentLinkedQueue<GlideImage> POOL = new ConcurrentLinkedQueue<>();

    public static GlideImage obtain() {
        GlideImage info = POOL.poll();
        return info != null ? info : new GlideImage();
    }

    public static void recycle(GlideImage info) {
        info.setWidth(0);
        info.setHeight(0);
        info.setResource(null);
        info.setResource(Resource.NONE);
        info.setFit(BoxFit.COVER);
        info.setScaleRatio(3.0);
        POOL.add(info);
    }
}

File: android/src/main/kotlin/org/axen/flutter/glide/common/GlideImagePool.kt
package org.axen.flutter.glide.common

import org.axen.flutter.glide.enum.BoxFit
import org.axen.flutter.glide.enum.Resource
import java.util.concurrent.ConcurrentLinkedQueue

object GlideImagePool {
    private val POOL: ConcurrentLinkedQueue<GlideImage> = ConcurrentLinkedQueue()

    fun obtain(): GlideImage {
        val info = POOL.poll()
        return info?: GlideImage()
    }

    fun recycle(info: GlideImage) {
        info.apply {
            width = 0
            height = 0
            resource = null
            resourceType = Resource.NONE
            fit = BoxFit.COVER
            scaleRatio = 3.0
            POOL.add(this)
        }
    }
}